platform/ios/src/MLNMapView.mm

```diff
@@ -6811,6 +6812,119 @@ - (void)updateHeadingForDeviceOrientation
     return MLNFeaturesFromMBGLFeatures(features);
 }
 
+// MARK: Managing Feature State
+
+- (void)setFeatureState:(id<MLNFeature>)feature sourceID:(NSString *)sourceID state:(NSDictionary<NSString *, id> *)state {
+    MLNAssertIsMainThread();
+
+    if (!feature || !sourceID || !state) {
+        return;
+    }
+
+    NSString *featureID = [NSString stringWithFormat:@"%@", feature.identifier];
+    [self setFeatureStateForSource:sourceID sourceLayer:nil featureID:featureID state:state];
+}
+
+- (void)setFeatureStateForSource:(NSString *)sourceID
+                    sourceLayer:(nullable NSString *)sourceLayerID
+                      featureID:(NSString *)featureID
+                           state:(NSDictionary<NSString *, id> *)state {
```

Comment:
Why do we need both `setFeatureStateForSource` and `setFeatureState`?

`setFeatureState` doesn't add much convenience other than setting `sourceLayer: nil` for you right?

I would just keep `setFeatureStateForSource` and rename to `setFeatureState`. Same goes for the `getter`.